testonly/hammer/record.go

```diff
@@ -0,0 +1,61 @@
+package hammer
+
+import (
+	"fmt"
+	"reflect"
+	"sync"
+
+	"github.com/golang/glog"
+	"github.com/google/trillian/types"
+)
+
+const (
+	// How many SMRs to hold on to.
+	smrCount = 30
+)
+
+// smrStash provides thread-safe access to an ordered, bounded queue of previously
+// witnessed SMRs
+type smrStash struct {
+	mu sync.RWMutex
+
+	// SMRs are arranged from later to earlier (so [0] is the most recent), and the
+	// discovery of new SMRs will push older ones off the end. No SMR for a revision
+	// will be stored more than once.
+	smr [smrCount]*types.MapRootV1
```

Comment:
I don't know if more refactoring is a waste of time, but this makes me think of 
https://golang.org/pkg/container/list/

Reply:
I don't want to spend more time on this than needed. If we were going to do other work on this stash then this might be worth changing over. Thanks for the pointer though.